src/main/capi/arrow-c.cpp

```diff
@@ -83,10 +83,21 @@ duckdb_state duckdb_query_arrow_array(duckdb_arrow result, duckdb_arrow_array *o
 	if (!wrapper->current_chunk || wrapper->current_chunk->size() == 0) {
 		return DuckDBSuccess;
 	}
-	ArrowConverter::ToArrowArray(*wrapper->current_chunk, reinterpret_cast<ArrowArray *>(*out_array), wrapper->options);
+	ArrowConverter::ToArrowArray(*wrapper->current_chunk, reinterpret_cast<ArrowArray *>(*out_array),
+	                             wrapper->result->client_properties);
```

Comment:
If you're going to use the wrapper->result, please check if it's not null first

Reply:
Since checking `wrapper->current-chunk` before implies that it was retrieved from the query result, I think we can expect `wrapper->result` not to be null

Reply:
I agree, but that's considering the current implementation, if this detail changes in the future this could lead to a crash